web/controllers/camera_controller.ex

```
@@ -210,4 +235,77 @@ defmodule EvercamMedia.CameraController do
+    camera_params = add_parameter("host", camera_params, "external_rtsp_port", params["external_rtsp_port"])
+
+    camera_params = add_parameter("host", camera_params, "internal_host", params["internal_host"])
+    camera_params = add_parameter("host", camera_params, "internal_http_port", params["internal_http_port"])
+    camera_params = add_parameter("host", camera_params, "internal_rtsp_port", params["internal_rtsp_port"])
+
+    camera_params = add_parameter("url", camera_params, "jpg", params["jpg_url"])
+    camera_params = add_parameter("url", camera_params, "mjpg", params["mjpg_url"])
+    camera_params = add_parameter("url", camera_params, "h264", params["h264_url"])
+    camera_params = add_parameter("url", camera_params, "audio", params["audio_url"])
+    camera_params = add_parameter("url", camera_params, "mpeg", params["mpeg_url"])
+
+    camera_params = add_parameter("auth", camera_params, "username", params["cam_username"])
+    camera_params = add_parameter("auth", camera_params, "password", params["cam_password"])
+
+    Camera.changeset(camera, camera_params)
+  end
+
+  defp add_parameter(_field, params, _key, value) when value in [nil, ""], do: params
+  defp add_parameter("field", params, key, value) do
+    Map.put(params, key, value)
+  end
+  defp add_parameter("model", params, key, value) do
+    Map.put(params, key, value.id)
+  end
+  defp add_parameter("host", params, key, value) do
+    put_in(params, [:config, key], value)
+  end
+  defp add_parameter("url", params, key, value) do
+    put_in(params, [:config, "snapshots", key], value)
+  end
+  defp add_parameter("auth", params, key, value) do
+    put_in(params, [:config, "auth", "basic", key], value)
+  end
+  defp add_parameter("location", params, _key, lng, lat) when lng in [nil, ""], do: params
+  defp add_parameter("location", params, key, lng, nil), do: Map.put(params, key, %Geo.Point{coordinates: {lng, 0}})
```

I think that in case of missing latitude or longitude we should return an error instead of defaulting to 0. It's very unlikely that a user would want that.